manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml

```diff
@@ -15,71 +39,151 @@ metadata:
     app: sidecar-injector
     release: {{ .Release.Name }}
 webhooks:
-  - name: sidecar-injector.istio.io
-    clientConfig:
-      {{- if .Values.istiodRemote.injectionURL }}
-      url: {{ .Values.istiodRemote.injectionURL }}
-      {{- else }}
-      service:
-        name: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
-        namespace: {{ .Release.Namespace }}
-        path: "/inject"
-      {{- end }}
-      caBundle: ""
-    sideEffects: None
-    rules:
-      - operations: [ "CREATE" ]
-        apiGroups: [""]
-        apiVersions: ["v1"]
-        resources: ["pods"]
-    failurePolicy: Fail
-    admissionReviewVersions: ["v1beta1", "v1"]
-    namespaceSelector:
-{{- if .Values.sidecarInjectorWebhook.enableNamespacesByDefault }}
-      matchExpressions:
-      - key: name
-        operator: NotIn
-        values:
-        - {{ .Release.Namespace }}
-      - key: istio-injection
-        operator: NotIn
-        values:
-        - disabled
-      - key: istio-env
-        operator: DoesNotExist
-      - key: istio.io/rev
-        operator: DoesNotExist
-{{- else if .Values.revision }}
-      matchExpressions:
-      - key: istio-injection
-        operator: DoesNotExist
-      - key: istio.io/rev
-        operator: In
-        values:
-        - {{ .Values.revision }}
+{{- if .Values.sidecarInjectorWebhook.useLegacySelectors}}
```

Comment:
Can we have the useLegacySelectors in mesh config ?
I don't remember where sidecarInjectWebhook is coming from...

Reply:
What is the benefit? I thought we decided that Kubernetes things would be configured by helm api, and runtime components by mesh config. This is directly configuring a k8s object (MutatingAdmissionWebhook).

sidecarInjectWebhook is in values.yaml

Additionally, this is a feature flag - not a proper API - and will be removed in 1.11